tuning/automation/rocblas-parser.py

```diff
@@ -0,0 +1,367 @@
+      previous_size = (0,0,0,0,0,0,0) if strided else (0,0,0,0,0,0)
+      for size in matrices[gemmtype]:
+        duplicate = ""
+        if size[:-2] != previous_size:
+            duplicate = "duplicate" if blas_layer == 2 else ""
+            count_unique+=1
+
+        if (strided): # for strided
+            gemmconfig = f" - [ {size[0]}, {size[1]}, {size[2]}, {size[3]} ] # lda: {size[4]} ldb: {size[5]} ldc: {size[6]} alpha: {size[7]} beta: {size[8]}"
+            if (blas_layer==4):
+                gemmconfig +=f" call_count: {size[9]}  {duplicate}\n"
+            else:
+                gemmconfig +=f"  {duplicate}\n"
+        else:
+            gemmconfig = f" - [ {size[0]}, {size[1]}, 1, {size[2]} ] # lda: {size[3]} ldb: {size[4]} ldc: {size[5]} alpha: {size[6]} beta: {size[7]}"
+            if (blas_layer==4):
+                gemmconfig +=f" call_count: {size[8]}  {duplicate}\n"
+            else:
+                gemmconfig +=f"  {duplicate}\n"
+
+        f.write(gemmconfig)
+        previous_size = size[:-2]
+
+    f.write(f"\n -- total unique is: {count_unique}")
+
+def create_rocBLAS_bench(benchFile, matrices,verify,initialization):
+
+    bench = []
+
+    for gemmtype in matrices:
+        print(gemmtype)
+        # get GEMM function type: a/b/c/d/compute_type
+        problemDict = {}
+        if ("HHS" in gemmtype):
+            problemDict["rocblas_function"] = "rocblas_gemm_strided_batched_ex" if "SB" in gemmtype else  "rocblas_gemm_ex"
+            problemDict["a_type"] = "f16_r"
+            problemDict["b_type"] = "f16_r"
+            problemDict["c_type"] = "f16_r"
+            problemDict["d_type"] = "f16_r"
+            problemDict["compute_type"] = "f32_r"
+    
+        elif ("HSS" in gemmtype):
+            problemDict["rocblas_function"] = "rocblas_gemm_strided_batched_ex" if "SB" in gemmtype else  "rocblas_gemm_ex"
+            problemDict["a_type"] = "f16_r"
+            problemDict["b_type"] = "f16_r"
+            problemDict["c_type"] = "f32_r"
+            problemDict["d_type"] = "f32_r"
+            problemDict["compute_type"] = "f32_r"
+
+        elif ("BBS" in gemmtype):
+            problemDict["rocblas_function"] = "rocblas_gemm_strided_batched_ex" if "SB" in gemmtype else  "rocblas_gemm_ex"
+            problemDict["a_type"] = "bf16_r"
+            problemDict["b_type"] = "bf16_r"
+            problemDict["c_type"] = "bf16_r"
+            problemDict["d_type"] = "bf16_r"
+            problemDict["compute_type"] = "f32_r"
+    
+        elif ("BSS" in gemmtype):
+            problemDict["rocblas_function"] = "rocblas_gemm_strided_batched_ex" if "SB" in gemmtype else  "rocblas_gemm_ex"
+            problemDict["a_type"] = "bf16_r"
+            problemDict["b_type"] = "bf16_r"
+            problemDict["c_type"] = "f32_r"
+            problemDict["d_type"] = "f32_r"
+            problemDict["compute_type"] = "f32_r"
+
+        elif ("SGEMM" in gemmtype):

```

Comment:
Can we always use _ex version?
Rotating buffer support is only with _ex.

Reply:
I'll add CGEMM and ZGEMM.